Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/Instance/Instance.h

```diff
@@ -80,182 +83,275 @@ namespace AzToolsFramework
+
+            //! Detaches an entity from the instance.
+            //! It unregisters the entity id and moves the entity ownership to the caller.
+            //! @param entityId The entity id to be detached.
+            //! @return Unique pointer to the entity.
             AZStd::unique_ptr<AZ::Entity> DetachEntity(const AZ::EntityId& entityId);
+
+            //! Detaches entities from the instance.
+            //! It unregisters the entity ids and moves the entity ownerships to the callback provided by the caller.
+            //! @param callback A user provided callback that can be used to capture ownership and manipulate the detached entities.
             void DetachEntities(const AZStd::function<void(AZStd::unique_ptr<AZ::Entity>)>& callback);
-            /**
-             * Replaces the entity stored under the provided alias with a new one.
-             *
-             * @return The original entity or a nullptr if not found.
-             */
-            AZStd::unique_ptr<AZ::Entity> ReplaceEntity(AZStd::unique_ptr<AZ::Entity>&& entity, EntityAliasView alias);
 
-            /**
-            * Detaches all entities in the instance hierarchy.
-            * Includes all direct entities, all nested entities, and all container entities.
-            * Note that without container entities the hierarchy that remains cannot be used further without restoring new ones.
-            * @param callback A user provided callback that can be used to capture ownership and manipulate the detached entities.
-            */
+            //! Detaches all entities in the instance hierarchy.
+            //! Includes all direct entities, all nested entities, and all container entities.
+            //! Note that without container entities the hierarchy that remains cannot be used further without restoring new ones.
+            //! @param callback A user provided callback that can be used to capture ownership and manipulate the detached entities.
             void DetachAllEntitiesInHierarchy(const AZStd::function<void(AZStd::unique_ptr<AZ::Entity>)>& callback);
 
-            void RemoveNestedEntities(const AZStd::function<bool(const AZStd::unique_ptr<AZ::Entity>&)>& filter);
+            //! Replaces the entity stored under the provided alias with a new one.
+            //! @param entity The new entity.
+            //! @param alias The alias of the entity that will be replaced.
+            //! @return The original entity or a nullptr if not found.
+            AZStd::unique_ptr<AZ::Entity> ReplaceEntity(AZStd::unique_ptr<AZ::Entity>&& entity, EntityAliasView alias);
 
-            void Reset();
+            //! Detaches entities in the instance hierarchy based on the provided filter.
+            //! @param filter A user provided filter used to decide whether an entity should be detached. Returns true to detach.
+            void RemoveEntitiesInHierarchy(const AZStd::function<bool(const AZStd::unique_ptr<AZ::Entity>&)>& filter);
```

Comment:
Should this be called RemoveAllEntitiesInHierarchy for consistency?

Reply:
I did not change this one because it does not remove all entities but remove entities based on a filter. I think this would better express what it is doing.

Reply:
Oh yeah, I thought it was just a callback. Never mind. :)